File: StudyNotes/GeekBang.org/视频课/算法面试通关40讲/06 | 面试题：反转一个单链表&判断链表是否有环/0206/206.cpp
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* prev = nullptr;
        ListNode* cur = head;
        while (cur != nullptr) {
            ListNode* tmp = cur->next;
            cur->next = prev;
            prev = cur;
            cur = tmp;
        }
        return prev;
    }
};

File: StudyNotes/GeekBang.org/视频课/算法面试通关40讲/06 | 面试题：反转一个单链表&判断链表是否有环/0206/206.swift
class Solution {
    func reverseList(_ head: ListNode?) -> ListNode? {
        var prev: ListNode?
        var cur = head
        while (cur != nil) {
            let tmp = cur?.next;
            cur?.next = prev
            prev = cur
            cur = tmp
        }
        return prev
    }
}
